### DISCUSSION TITLE
Improvement of hands up animation?

### DISCUSSION THREAD

**Discussion by alberttheprince** [alberttheprince/rpemotes-reborn#53] [Ideas]
Just a suggestion, as the current "snappy" hands up animation doesn't look visually nice.

It might be preferable to use the anim/dict: "random@mugging5", "ig_2_guy_enter_handsup" as it provides a more "seamless" hands up experience.

This is code I used for hands up a while back:

https://github.com/user-attachments/assets/da032418-88ba-42f4-bc7b-540a25cf63df

```
local function handleHandsup(bool)
    lib.requestAnimDict("random@mugging5")
    handsup = bool
    if handsup then
        LocalPlayer.state.invHotkeys = false
        TaskPlayAnim(cache.ped, "random@mugging5", "ig_2_guy_enter_handsup", 0.5, 0.5, -1, 50, 0, false, false, false)
        exports['qb-smallresources']:addDisableControls(Config.disableHandsupControls)
    else
        TaskPlayAnim(cache.ped, "random@mugging5", "ig_2_guy_exit_handsup", 0.5, 0.5, -1, 26, 0, false, false, false)
        Wait(5)
        ClearPedTasks(cache.ped)
        exports['qb-smallresources']:removeDisableControls(Config.disableHandsupControls)
        Wait(1000)
        LocalPlayer.state.invHotkeys = true
    end
end

**Comment by iSentrie** [comment]
Or just add hands down exits?
```lua
RP.Exits = {
    ["handsdown"] = {
        "random@mugging5",
        "ig_2_guy_exit_handsup",
        "Hands down",
        AnimationOptions = {
            EmoteDuration = 2250,
            StartDelay = 600,
            EmoteMoving = true
        }
    },
},

RP.Emotes = {
    ['handsup6'] = {
        'random@mugging5',
        'ig_2_guy_enter_handsup',
        'Hands Up 6',
        AnimationOptions = {
            EmoteMoving = true,
            EmoteLoop = false,
            ExitEmote = "handsdown",
            ExitEmoteType = "Exits"
        }
    }
}
```

**Reply by alberttheprince** [reply]
Sure, that's good! This might be good for all the hands up updates honestly, the current "snap" to normal is pretty jarring.

**Reply by alberttheprince** [reply]
@iSentrie Would you be also updating the hands up script to handsup6? Seems like it would just be a general upgrade (even with the issue of that 'jitter' when going down) for everyone.

**Reply by iSentrie** [reply]
> @iSentrie Would you be also updating the hands up script to handsup6? Seems like it would just be a general upgrade (even with the issue of that 'jitter' when going down) for everyone.

not really, it loops for me even with `EmoteLoop = false` , i added exits for few others too, everything looks bad honestly. see details in discord
